contracts/NonfungiblePositionManager.sol

```diff
@@ -106,34 +94,54 @@ contract NonfungiblePositionManager is INonfungiblePositionManager, BaseLiquidit
         uint256 liquidity,
         uint256 deadline,
         address owner,
-        bytes memory hookData,
-        address sender
-    ) internal checkDeadline(deadline) returns (BalanceDelta delta, uint256 tokenId) {
-        delta = _increaseLiquidity(owner, range, liquidity, hookData, sender);
-
+        bytes memory hookData
+    ) internal checkDeadline(deadline) returns (BalanceDelta delta) {
         // mint receipt token
-        _mint(owner, (tokenId = nextTokenId++));
+        uint256 tokenId;
+        unchecked {
+            tokenId = nextTokenId++;
+        }
+        _mint(owner, tokenId);
+
+        (delta,) = _modifyLiquidity(range, liquidity.toInt256(), bytes32(tokenId), hookData);
+
         tokenPositions[tokenId] = TokenPosition({owner: owner, range: range});
     }
 
-    function increaseLiquidity(uint256 tokenId, uint256 liquidity, bytes memory hookData, bool claims, address sender)
+    // Note: Calling increase with 0 will accrue any underlying fees.
+    function increaseLiquidity(uint256 tokenId, uint256 liquidity, bytes memory hookData, address sender)
         internal
         isAuthorizedForToken(tokenId, sender)
         returns (BalanceDelta delta)
     {
         TokenPosition memory tokenPos = tokenPositions[tokenId];
-
-        delta = _increaseLiquidity(tokenPos.owner, tokenPos.range, liquidity, hookData, sender);
+        // Note: The tokenId is used as the salt for this position, so every minted liquidity has unique storage in the pool manager.
+        (delta,) = _modifyLiquidity(tokenPos.range, liquidity.toInt256(), bytes32(tokenId), hookData);
     }
 
-    function decreaseLiquidity(uint256 tokenId, uint256 liquidity, bytes memory hookData, bool claims, address sender)
+    // Note: Calling decrease with 0 will accrue any underlying fees.
+    function decreaseLiquidity(uint256 tokenId, uint256 liquidity, bytes memory hookData, address sender)
         internal
         isAuthorizedForToken(tokenId, sender)
         returns (BalanceDelta delta)
     {
         TokenPosition memory tokenPos = tokenPositions[tokenId];
+        (delta,) = _modifyLiquidity(tokenPos.range, -(liquidity.toInt256()), bytes32(tokenId), hookData);
+    }
 
-        delta = _decreaseLiquidity(tokenPos.owner, tokenPos.range, liquidity, hookData);
+    // there is no authorization scheme because the payer/recipient is always the sender
+    // TODO: Add more advanced functionality for other payers/recipients, needs auth scheme.
```

Comment:
yeah i feel like it would be particularly nice for `take` to be able to have a `recipient` and theres no security problems there right